File: src/jvmTest/kotlin/community/flock/kotlinx/rgxgen/data/DataInterface.java
package community.flock.kotlinx.rgxgen.data;

import community.flock.kotlinx.rgxgen.nodes.Node;

import java.util.List;

public interface DataInterface {

    boolean hasEstimatedCount();

    boolean hasAllUniqueValues();

    boolean useFindForMatching();

    boolean isUsableWithJavaPattern();

    String getPattern();

    Node getResultNode();

    Long getEstimatedCount();

    List<String> getAllUniqueValues();
}

File: src/jvmTest/kotlin/community/flock/kotlinx/rgxgen/data/DataInterface.kt
package community.flock.kotlinx.rgxgen.data

import community.flock.kotlinx.rgxgen.nodes.Node

interface DataInterface {
    fun hasEstimatedCount(): Boolean

    fun hasAllUniqueValues(): Boolean

    fun useFindForMatching(): Boolean

    fun isUsableWithJavaPattern(): Boolean

    fun getPattern(): String?

    fun getResultNode(): Node?

    fun getEstimatedCount(): Long?

    fun getAllUniqueValues(): List<String?>?
}
